core/App.php

```diff
@@ -274,13 +274,7 @@ public static function checkout_process($orderId){
             }
 
             if (array_key_exists('cmw_register_email', $_POST)){
-                if (!empty($email)){

```

Comment:
Essentially, previously there is this 2 ifs statement --> had combined them and only actioned if the end user select the option as true.